lib/osf-components/addon/components/contributor-list/component.ts

```diff
@@ -1,29 +1,123 @@
+export default class ContributorList extends Component.extend({
+    loadContributors: task(function *(this: ContributorList, more?: boolean) {
+        if (!this.node) {
+            return;
+        }
+
+        // we are running in Prerender, load and show all contributors at once.
+        if (this.userIsBot) {
+            const botBlocker = this.ready.getBlocker();
+
+            const allContributors = yield this.node.loadAll('contributors');
+            this.set('contributors', allContributors);
+            this.set('displayedContribs', allContributors.length);
+
+            botBlocker.done();
+            return;
+        }
+
+        const contributors = yield this.node.hasMany('contributors');
+        if (!more && contributors) {
+            this.set('contributors', contributors.value());
+            return;
+        }
+
+        if (more) {
+            // we have enough contributors to show, don't request any more.
+            const hiddenContribs = this.contributorList.length - this.displayedContribs;
+            if (this.rest <= hiddenContribs) {
+                if (this.useShowStep) {
+                    this.incrementProperty('displayedContribs', this.showStep);
+                } else {
+                    this.set('displayedContribs', this.contributorList.length);
+                }
+                return;
+            }
+        }
+
+        const blocker = this.ready.getBlocker();
+
+        // Fetch more contributors.
+        const contribs = yield this.node.queryHasMany('contributors', {
+            page: more ? this.incrementProperty('page') : this.set('page', 1),
+            'page[size]': this.contribsPerPage,
+        });
+
+        if (more && this.contributors) {
+            this.contributors.pushObjects(contribs);
+            if (this.useShowStep) {
+                this.incrementProperty('displayedContribs', this.showStep);
+            } else {
+                this.set('displayedContribs', this.contributorList.length);
+            }
+        } else {
+            this.set('contributors', contribs);
+        }
+
+        blocker.done();
+    }).restartable(),
+}) {
     layout = layout;
+    styles = styles;
 
     @service i18n!: I18N;
-    max = 3;
-    nodeId?: string;
-    useLink?: boolean;
-    contributors!: DS.PromiseManyArray<Contributor> & { meta: { total: number } };
+    @service store!: DS.Store;
+    @service ready!: Ready;
+
+    page = 1;
+    defaultStep = 3;
+    userIsBot: boolean = navigator.userAgent.includes('Prerender');
+    contributors?: QueryHasManyResult<Contributor>;
+
+    // Required arguments
+    node!: Node;
+
+    // Optional arguments
+    showStep?: number = defaultTo(this.showStep, 0);
+    contribsPerPage?: number = defaultTo(this.contribsPerPage, 10);
+    displayedContribs: number = defaultTo(this.displayedContribs, this.defaultStep);
+    showLoading: boolean = defaultTo(this.showLoading, true);
+    darkBall: boolean = defaultTo(this.darkBall, true);
```

Comment:
It's nice to keep a somewhat consistent interface across our components.
Chances are that if you're opting for a light or dark loading indicator you'll want the rest of the component to follow in suit.

```suggestion
    dark: boolean = defaultTo(this.dark, true);
```